src/views/DraftDetail.vue

```diff
@@ -342,24 +381,63 @@ async function findProduct() {
   isSearchingProduct.value = false
 }
 
-async function addProductToCount() {
-  // If product is not found in the searched string then do not make the api call
-  // This check is only required to handle the case when user presses the enter key on the input and we do not have any result in the searchedProduct
-  if(!searchedProduct.value.productId || !queryString.value) {
-    return;
+async function findProductFromIdentifier(payload: any) {
+  
+  const idType = payload.idType;
+  const idValues = payload.idValue;
+  const filterString = `${idType}: (${idValues.join(' OR ')})`;
+
+  try {
+    const resp = await ProductService.fetchProducts({
+      "filters": [filterString],
+      "viewSize": idValues.length
+    })
+
+    if (!hasError(resp) && resp.data.response?.docs?.length) {
+      const products = resp.data.response.docs;
+      const itemsAlreadyInCycleCount = currentCycleCount.value.items.map((item: any) => item.productId);
+
+      // Filter payload products to only include items that have a matching product and are not already in the cycle count.
+      const filteredProducts = products.filter((product: any) => {
+        return idValues.includes(product[idType]) && !itemsAlreadyInCycleCount.includes(product.productId);
+      });
+
+      const productToAdd = filteredProducts.map((product: any) => ({ idType, idValue: product[idType] }));
+
+      await addProductToCount(productToAdd, true);
```

Comment:
If no products are available to be added then display an error and do not make call for add item api